apps/emqx_dashboard/src/emqx_dashboard_swagger.erl

```diff
@@ -152,6 +153,30 @@ spec(Module, Options) ->
         ),
     {ApiSpec, components(lists:usort(AllRefs), Options)}.
 
+is_content_type_json(Params, Meta) ->
+    validate_content_type(Params, Meta, <<"application/json">>).
+
+%% tip: Skip content-type check if body is empty.
+validate_content_type(#{body := Body} = Params, #{method := Method}, Expect) when
+    (Method =:= put orelse
+        Method =:= post orelse
+        method =:= patch) andalso
+        Body =/= #{}
+->
+    ExpectSize = byte_size(Expect),
+    case find_content_type(Params) of
+        <<Expect:ExpectSize/binary, _/binary>> ->
+            {ok, Params};
+        _ ->
+            {415, 'UNSUPPORTED_MEDIA_TYPE', <<"content-type:", Expect/binary, " Required">>}
+    end;
+validate_content_type(Params, _Meta, _Expect) ->
+    {ok, Params}.
+
+find_content_type(#{headers := #{<<"content-type">> := Type}}) -> Type;
+find_content_type(#{headers := #{<<"Content-Type">> := Type}}) -> Type;
+find_content_type(_Headers) -> error.
```

Comment:
HTTP RFC declares that header names are content-insensitive, this check should take it into account.

Perhaps, it can handle happy cases as implemented below, and in this clause run a more comprehensive check for pathological cases like `CoNtEnT-TyPe`.

> HTTP headers let the client and the server pass additional information with an HTTP request or response. An HTTP header consists of its case-insensitive name followed by a colon (:), then by its value. [Whitespace](https://developer.mozilla.org/en-US/docs/Glossary/Whitespace) before the value is ignored.

Reply:
I think cowboy normalizes header names to lower-case:
https://github.com/emqx/cowboy/blob/957dc318057b1adf1c57c021cd097d2820a13880/src/cowboy_http.erl#L605

Reply:
Makes sense. Then second clause is not needed.